src/mame/ddr/kc_m.cpp

```diff
@@ -593,9 +593,37 @@ void kc_state::pio_portb_w(uint8_t data)
 
 	update_0x08000();
 
-	/* 16 speaker levels */
-	m_speaker_level = (data>>1) & 0x0f;
+	// KC 85/2..3: 5-bit DAC
+	m_dac_level = (~data & 0x1f)>>1;
```

Comment:
Is this shift correct? The `>>1` seems to be a typo here, considering that it is supposed to be a 5-bit DAC.

Maybe you can just write the function once and have `kc85_3_state::pio_portb_w` fall back to `kc_state::pio_portb_w`.

Reply:
Whoops! You are right, that shift is incorrect and removing it obsoletes `kc85_3_state::pio_portb_w`. Thanks for spotting this, I'll correct it in an upcoming PR.